### PR TITLE
Integrate Secret Scanning from `Microsoft.Security.Utilities`

### PR DESCRIPTION
**Pull request by scbedd**
lol_parallel_is_the_way

todo:

- [x] Make the call as to scanning entire repo vs only changed files
- [x] Better reporting of secret scanner errors
- [x] Add CLI Integration test which fakes a real test secret and prevents `push`
- [x] Additional styling for clarity of communication. Add Line # to detection.

### CODE REVIEW

File: tools/test-proxy/Azure.Sdk.Tools.TestProxy/Common/SecretScanner.cs [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+            Console = consoleWrapper;
+        }
+
+        public async Task<List<Tuple<string, Detection>>> DiscoverSecrets(string assetRepoRoot, IEnumerable<string> relativePaths)
+        {
+            var detectedSecrets = new List<Tuple<string, Detection>>();
+            var total = relativePaths.Count();
+            var seen = 0;
+            Console.WriteLine(string.Empty);
+            foreach (string filePath in relativePaths)
```
Comment (benbp) [inline tools/test-proxy/Azure.Sdk.Tools.TestProxy/Common/SecretScanner.cs:31]: Is this a reasonably quick operation or does it make sense to parallelize?

Comment (scbedd) [reply tools/test-proxy/Azure.Sdk.Tools.TestProxy/Common/SecretScanner.cs:31]: For that example in the gif, it runs in about 20 seconds. Most won't be doing that.

I'll convert this to Parallel task invocation. NBD.

**Comment by weshaggard** [comment]
FYI @michaelcfanning

File: tools/test-proxy/Azure.Sdk.Tools.TestProxy/Console/ConsoleWrapperTester.cs [hunk 2]
```diff
@@ -1,4 +1,4 @@
-﻿using System;
+using System;
```
Comment (scbedd) [inline tools/test-proxy/Azure.Sdk.Tools.TestProxy/Console/ConsoleWrapperTester.cs:1]:
```suggestion
```

File: tools/test-proxy/Azure.Sdk.Tools.TestProxy/Console/ConsoleWrapper.cs [hunk 3]
```diff
@@ -1,4 +1,4 @@
-﻿
+
 using System;
```
Comment (scbedd) [inline tools/test-proxy/Azure.Sdk.Tools.TestProxy/Console/ConsoleWrapper.cs:2]:
```suggestion
```

**Pull request merged**